### PR TITLE
Vectorize calculating iou with a batch of Box2d

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: python/rikai/types/geometry.py [hunk 1]
```diff
@@ -263,22 +263,40 @@ def area(self) -> float:
...
         return self.width * self.height
 
+    @staticmethod
+    def _area(arr: np.ndarray) -> np.array:
+        """Calculate box area from an array of boxes."""
+        return np.maximum(0, arr[:, 2] - arr[:, 0]) * np.maximum(
+            0, arr[:, 3] - arr[:, 1]
+        )
+
     def iou(self, other: "Box2d") -> float:
```
Comment (changhiskhan) [inline python/rikai/types/geometry.py:273]: update type hints?

File: python/rikai/types/geometry.py [hunk 2]
```diff
@@ -263,22 +263,40 @@ def area(self) -> float:
...
         """Compute intersection over union(IOU)."""
         assert isinstance(
-            other, Box2d
+            other, (Box2d, list)
```
Comment (changhiskhan) [inline python/rikai/types/geometry.py:276]: what about like a tuple or array of Box2d?

**Review by changhiskhan — APPROVED** [review]
2 nits

File: python/rikai/types/geometry.py (hunk 2 above)
Comment (eddyxu) [reply python/rikai/types/geometry.py]: done

**Pull request merged**